src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift

```diff
@@ -260,27 +265,46 @@ extension RiskProvider: RiskProviding {
 		let group = DispatchGroup()
 
 		group.enter()
-		var appConfiguration: SAP_ApplicationConfiguration?
+
 		appConfigurationProvider.appConfiguration { [weak self] result in
 			switch result {
 			case .success(let config):
-				appConfiguration = config
-
-				self?.determineSummary(userInitiated: userInitiated, ignoreCachedSummary: ignoreCachedSummary, appConfiguration: config) {
-					summary = $0
-					group.leave()
-				}
+				self?.determineSummary(
+					userInitiated: userInitiated,
+					ignoreCachedSummary: ignoreCachedSummary,
+					appConfiguration: config,
+					completion: { result in
+						guard let self = self else { return }
```

Comment:
Better guard let self on start- so you can also eliminate more self?. ?

Reply:
done